### ISSUE TITLE
tslib v1.13.0 has breaking change

### ISSUE THREAD

**Issue by MSNev** [microsoft/ApplicationInsights-JS#1269]
tslib v 1.13.0 is causing a compile error due to a breaking change

Compiling...
Using tsc v2.5.3
common/temp/node_modules/tslib/tslib.d.ts(37,78): error TS2304: Cannot find name 'PropertyKey'.
common/temp/node_modules/tslib/tslib.d.ts(37,103): error TS2304: Cannot find name 'PropertyKey'.

We have created an issue with tslib to try and get this fixed, if not we will need to roll back and be fixed to v 1.11.2. As a temporary workaround if you have this issue is to update your local package.json files to reference "1.11.2" instead of "^1.11.1"

Raised TSLib Issue
https://github.com/microsoft/tslib/issues/115

**Comment by MSNev** [comment]
As this is the 2nd time in recent history and looking at how the tslib project is now structured, it appears that it's going to remain busted for us as we are currently targeting es3 as our base module to ensure that we don't accidently break our SDK user base that has users that are currently still relying on ES3 based browsers.

As such, I'm going to completely break our dependency on tslib, without the normal corresponding code bloat (as size is still a major concern). But I'm attempting to do this in a backward compatible way so that we don't break
- CDN only users
- Older TypeScript users ( < 3.x)
- More modern TypeScript users (i.e. 3.x)
- Anyone how is currently user *any* version of tslib 
  - as previously we broke users using TypeScript and a newer version of tslib then we needed to be pinned to (v1.10.1 from memory
  - This is also a major consideration for breaking the dependency, so we don't break anyone who wants/needs to use a version of tslib > 1.11.2 (the last known good version that we can compile with)

**Comment by nicolashenry** [comment]
@MSNev Could you release a version still using tslib at least for esm version? The recent change create pollution in global object.

**Comment by MSNev** [comment]
Created #1280 to investigate how we could possibly stop exposing the global functions.

**Comment by nicolashenry** [comment]
> Is it just the __extends() or __assign() or are you seeing something else in the global scope

Both __extends and __assign but nothing else.

> Which modules / files are you ingesting (using) [i.e. just the AISku or the individual components, if the later which ones).

I am only using "@microsoft/applicationinsights-web" module (and indirectly its dependencies).

I would prefer tslib version to be fixed to 1.11.2 even if it would potentially create duplicated version of tslib in my projects because, since I am using rollupjs, it would be the same result as with 2.5.5 except that it would not pollute the global object (but I understand that it could cause some trouble in a different configuration than mine to have different versions of tslib in the same project).

I see your issue here https://github.com/microsoft/tslib/issues/115 , if the problem is only with "es3" target in a major version, I would suggest to drop support for this target and change it to "es5" (all alive browsers including IE11 support it since more than 10 years now).

**Comment by MSNev** [comment]
Looking at the generated code the /dist/ folder looks like we *could* safely stop exposing these methods, but, for the /dist-esm/ we can't, without declaring this as a breaking change and requiring users to provide their own versions of these methods.

Unfortunately, we can't just fix it @ 1.11.2 as this breaks some consumers such as the discussion in #1242. So that's not really an option.
 
Likewise we also have users who's end-users are still using IE7/IE8 (i.e. medical organizations that are still using XP), so while it would be *great* to be able to re-target @ es5 we would also need to beef up our testing infrastructure to ensure that we don't break any es3 users.... However, our current test infra actually doesn't work on IE at all let alone IE running in IE7/8 mode.

So for now that is not *currently* an option either...

On the upside, these methods are only exposed via the applicationinsights-shims module, so we only have a single location to make the change (once we identify a safe way to achieve this).

**Comment by nicolashenry** [comment]
You may can re-bind "tslib" to a local module with some post-build script after typescript transpilation. For example:
```ts
import * as tslib_1 from "tslib";
```
to
```ts
import * as tslib_1 from "./tslib.js";
```
or maybe bind it to your own tslib compatible module:
```ts
import * as tslib_1 from "@microsoft/applicationinsights-shims";
```
It would maybe break sourcemaps however.

Thanks for your answers :).

**Comment by MSNev** [comment]
Included in v 2.5.5 which is now fully deployed
